source/proximity_list/proximity_list.cpp

```diff
@@ -1,9 +1,158 @@
+ * (1%), the approximation gives quite good result.( (sin(0.01rad) - 0.01) < 1.7e-7 ! )
+ *
+ * Otherwise, the correction factor is computed (only when the search radius > 1%*Earth's radius)
+ *
+ * */
+static float search_radius_correction_factor(const double& radius) {
+    return radius < GeographicalCoord::EARTH_RADIUS_IN_METERS * 0.01
+               ? 1.f
+               : 2 * GeographicalCoord::EARTH_RADIUS_IN_METERS
+                     * asin(radius / (2.f * GeographicalCoord::EARTH_RADIUS_IN_METERS)) / radius;
+}
+
+template <class T>
+void ProximityList<T>::build() {
+    log4cplus::Logger logger = log4cplus::Logger::getInstance("log");
+    LOG4CPLUS_INFO(logger, "Building Proximitylist's NN index with " << items.size() << " items");
+
+    // clean NN index
+    NN_data.clear();
+    NN_index.reset();
+
+    if (items.empty()) {
+        LOG4CPLUS_WARN(logger, "No items for building the index");
+        return;
+    }
+
+    for (const auto& i : items) {
+        auto projected = project_coord(i.coord);
+        std::copy(projected.begin(), projected.end(), std::back_inserter(NN_data));
+    }
+    auto points = flann::Matrix<float>{&NN_data[0], NN_data.size() / 3, 3};
+    NN_index = std::make_shared<navitia::proximitylist::index_t>(points, flann::KDTreeSingleIndexParams(10));
+    NN_index->buildIndex();
+}
+
+template <typename T, typename Item>
+static auto extract(const Item& item, IndexCoord) -> typename ReturnTypeTrait<T, IndexCoord>::ValueType {
+    return std::make_pair(item.element, item.coord);
+}
+
+template <typename T, typename Item>
+static auto extract(const Item& item, IndexOnly) -> typename ReturnTypeTrait<T, IndexOnly>::ValueType {
+    return item.element;
+}
+
+template <typename T, typename Items, typename Indices, typename Distances, typename Tag>
+static auto make_result(const type::GeographicalCoord& coord,
+                        const Items& items,
+                        const Indices& indices,
+                        const Distances& distances,
+                        const int nb_found,
+                        Tag) -> std::vector<typename ReturnTypeTrait<T, Tag>::ValueType> {
+    log4cplus::Logger logger = log4cplus::Logger::getInstance("log");
+
+    std::vector<typename ReturnTypeTrait<T, Tag>::ValueType> res;
+    if (!nb_found) {
+        LOG4CPLUS_TRACE(log4cplus::Logger::getInstance("log"),
+                        "0 point found for the coord: " << coord.lon() << " " << coord.lat());
+        return {};
+    }
+    for (int i = 0; i < nb_found; ++i) {
+        int res_ind = indices[i];
+        if (res_ind < 0 || res_ind >= static_cast<int>(items.size())) {
+            continue;
+        }
+        LOG4CPLUS_TRACE(
+            log4cplus::Logger::getInstance("log"),
+            "Distance(squared) from the coord: " << coord.lon() << " " << coord.lat() << " is " << distances[i]);
+
+        res.push_back(std::move(extract<T>(items[res_ind], Tag{})));
```

Comment:
Can't you just use `res.emplace_back` instead and remove the `std::move` ?

Reply:
i think i can even just remove std::move.... `push_back` has an interface for rvalue actually....